### PR TITLE
fix: Make search feature in twitter client works

### PR DESCRIPTION
**Pull request by nulLeeKH**
# Relates to:

This PR has written without issue ticket

# Risks

Low

# Background

## What does this PR do?

Modified the enableSearch feature of the twitter client to make it work.

## What kind of change is this?

Bug fixes (non-breaking change which fixes an issue)

## Why are we doing this? Any context or related work?

We wanted the AI Agent to be exposed to random posts to learn and to respond to members of the community, and we found the TWITTER_SEARCH_ENABLE environment variable while looking for a way to do so. However, we confirmed that it was missing during the development process and that the environment variable was not working properly, so we modified it to be as identical to other toggle functions as possible.

# Documentation changes needed?

My changes do not require a change to the project documentation.

# Testing

## Where should a reviewer start?

Just run agent with twitter client with TWITTER_SEARCH_ENABLE environment variable

## Detailed testing steps

None, not much details needed.

## Discord username

blairlee_dev

### CODE REVIEW

File: agent/src/index.ts [hunk 1]
```diff
@@ -433,30 +429,32 @@ export async function initializeClients(
...
-        "off",
-        "null",
-        "undefined",
-        "",
-    ];
-
-    // Check if the value (trimmed and lowercased) is in the falsish list
-    return falsishValues.includes(value.trim().toLowerCase());
-}
+// no needed for now
```
Comment (monilpat) [inline agent/src/index.ts:432]: If it isn't needed can we remove it?

File: packages/client-twitter/src/index.ts [hunk 2]
```diff
@@ -34,12 +34,15 @@ export const TwitterClientInterface: Client = {
 
         elizaLogger.log("Twitter client started");
 
-        const manager = new TwitterManager(runtime, this.enableSearch);
+        const manager = new TwitterManager(runtime, runtime.getSetting("TWITTER_SEARCH_ENABLE") === "true");
```
Comment (monilpat) [inline packages/client-twitter/src/index.ts:37]: you should do a case insensitive comparison so lowercasing the returned setting

**Review by monilpat — CHANGES_REQUESTED** [review]
Thanks for doing this please address the comments then good to go :)

File: agent/src/index.ts (hunk 1 above)
Comment (nulLeeKH) [reply agent/src/index.ts:432]: removed on 510877303f244f2b0c6afdb8e350f8b0fb4606b0

File: packages/client-twitter/src/index.ts (hunk 2 above)
Comment (nulLeeKH) [reply packages/client-twitter/src/index.ts:37]: use toLowerCase() method on 510877303f244f2b0c6afdb8e350f8b0fb4606b0

**Review by monilpat — APPROVED** [review]
Thanks for addressing comments :)

**Review by odilitime — APPROVED** [review]
Much more elegant, banging my head, why didn't I think of that

**Pull request merged**